homeassistant/components/statistics/sensor.py

```diff
@@ -480,3 +435,165 @@ async def _initialize_from_database(self):
         self.async_schedule_update_ha_state(True)
 
         _LOGGER.debug("%s: initializing from database completed", self.entity_id)
+
+    def _update_attributes(self):
+        """Calculate and update the various attributes."""
+        self.attributes[STAT_BUFFER_USAGE_RATIO] = round(
+            len(self.states) / self._samples_max_buffer_size, 2
+        )
+
+        if len(self.states) >= 1 and self._samples_max_age is not None:
+            self.attributes[STAT_AGE_COVERAGE_RATIO] = round(
+                (self.ages[-1] - self.ages[0]).total_seconds()
+                / self._samples_max_age.total_seconds(),
+                2,
+            )
+        else:
+            self.attributes[STAT_AGE_COVERAGE_RATIO] = STATE_UNKNOWN
+
+    def _update_value(self):
+        """Front to call the right statistical characteristics functions.
+
+        One of the _stat_*() functions is represented by self._state_characteristic_fn().
+        """
+
+        if self.is_binary:
+            self._value = len(self.states)
+            return
+
+        value = self._state_characteristic_fn()
+
+        if self._state_characteristic not in STATS_NOT_A_NUMBER:
+            with contextlib.suppress(TypeError):
+                value = round(value, self._precision)
+                if self._precision == 0:
+                    value = int(value)
+        self._value = value
+
+    def _stat_average_linear(self):
+        if len(self.states) >= 2:
+            area = 0
+            for i in range(1, len(self.states)):
+                area += (
+                    0.5
+                    * (self.states[i] + self.states[i - 1])
+                    * (self.ages[i] - self.ages[i - 1]).total_seconds()
+                )
+            age_range_seconds = (self.ages[-1] - self.ages[0]).total_seconds()
+            return area / age_range_seconds
+        return STATE_UNKNOWN
```

Comment:
Why are we returning `STATE_UNKNOWN` instead of `None`?  We use `None` to represent unknown state. It will be translated in the base entity class to `STATE_UNKNOWN`.

Reply:
Hey Martin, I'm still learning best practices. So far I was not aware which is better and what the logical role of `None` is in Home Assistant.
Happy to change in a PR. @emontnemery fyi

Reply:
The entities in the platforms should return None to represent unknown state. The base entity class takes care of turning None into the correct string state:

https://github.com/home-assistant/core/blob/de1527d0e9b3b4e98925eb37ba83b5473d456796/homeassistant/helpers/entity.py#L487-L488

Reply:
@MartinHjelmare is 100% correct, I missed that.

Reply:
https://github.com/home-assistant/core/pull/60324